signer/tests/integration/postgres.rs

```diff
@@ -5580,3 +5580,1200 @@ async fn is_withdrawal_inflight_catches_withdrawals_in_package() {
+        let requests = db
+            .get_pending_withdrawal_requests(&bitcoin_block.block_hash, 1_000, &Faker.fake())
+            .await
+            .expect("failed to query db");
+        assert_eq!(requests.len(), 1);
+
+        // But it should not be "pending accepted".
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                &bitcoin_block.block_hash,
+                &stacks_block.block_hash,
+                min_block_height,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert!(requests.is_empty());
+
+        signer::testing::storage::drop_db(db).await;
+    }
+
+    /// Asserts that we only return requests that have been accepted by the
+    /// required number of signers. This test uses an accept threshold of `2`.
+    /// We create two withdrawal requests, one with two yes votes and one with
+    /// one yes vote and one no vote and ensure that only the request with two
+    /// yes votes is returned.
+    ///
+    /// This test creates blockchains with the following structure:
+    ///
+    /// ```text
+    ///          ┌────────┐
+    /// Bitcoin: │   B1 ◆ │
+    ///          └─▲──────┘  We both confirm (✔) and sweep (◆) the
+    ///          ┌─┴──────┐  request in B1.
+    /// Stacks:  │   S1 ✔ │
+    ///          └────────┘
+    /// ```
+    #[tokio::test]
+    async fn returns_only_withdrawals_accepted_by_threshold_signers() {
+        let db = signer::testing::storage::new_test_database().await;
+
+        let signature_threshold = 2;
+        let min_block_height = 0;
+
+        // Bitcoin blocks:
+        let bitcoin_block = BitcoinBlock::new_genesis();
+        // Stacks blocks:
+        let stacks_block = StacksBlock::new_genesis().anchored_to(&bitcoin_block);
+
+        // Write our blocks.
+        db.write_blocks([&bitcoin_block], [&stacks_block]).await;
+
+        // Setup withdrawal request 1: no votes
+        store_withdrawal_request(&db, 1, &bitcoin_block, &stacks_block, &[]).await;
+        // Setup withdrawal request 2: two yes votes
+        let withdrawal_request_2 =
+            store_withdrawal_request(&db, 2, &bitcoin_block, &stacks_block, &[true, true]).await;
+        // Setup withdrawal request 3: one yes vote, one no vote
+        store_withdrawal_request(&db, 3, &bitcoin_block, &stacks_block, &[true, false]).await;
+
+        // Get the pending accepted withdrawal requests. We should only
+        // get withdrawal_request2.
+        let mut requests = db
+            .get_pending_accepted_withdrawal_requests(
+                &bitcoin_block.block_hash,
+                &stacks_block.block_hash,
+                min_block_height,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+
+        assert_eq!(requests.len(), 1);
+        assert_eq!(requests.pop().unwrap(), withdrawal_request_2);
```

Comment:
nit: there's a very cool `.single()` we can use in those cases 😄

Reply:
Oh, right 😅

Reply:
[eceeb7f](https://github.com/stacks-network/sbtc/pull/1397/commits/eceeb7f6db1f6392f9363e0135f883301a7241d4)